cpu/kinetis_common/periph/pwm.c

```diff
@@ -24,375 +24,108 @@
  * @}
  */
 
-#include <stdint.h>
-#include <string.h>
-
 #include "cpu.h"
+#include "assert.h"
 #include "periph/pwm.h"
-#include "periph_conf.h"
 
-/* FTM channel look up tables */
-#if PWM_0_EN
-static const uint8_t ftm0chan[] = {
-#if PWM_0_CHANNELS > 0
-    PWM_0_CH0_FTMCHAN,
-#endif
-#if PWM_0_CHANNELS > 1
-    PWM_0_CH1_FTMCHAN,
-#endif
-#if PWM_0_CHANNELS > 2
-    PWM_0_CH2_FTMCHAN,
-#endif
-#if PWM_0_CHANNELS > 3
-    PWM_0_CH3_FTMCHAN,
-#endif
-#if PWM_0_CHANNELS > 4
-    PWM_0_CH4_FTMCHAN,
-#endif
-#if PWM_0_CHANNELS > 5
-    PWM_0_CH5_FTMCHAN,
-#endif
-#if PWM_0_CHANNELS > 6
-    PWM_0_CH6_FTMCHAN,
-#endif
-#if PWM_0_CHANNELS > 7
-    PWM_0_CH7_FTMCHAN,
-#endif
-};
-#endif
-#if PWM_1_EN
-static const uint8_t ftm1chan[] = {
-#if PWM_1_CHANNELS > 0
-    PWM_1_CH0_FTMCHAN,
-#endif
-#if PWM_1_CHANNELS > 1
-    PWM_1_CH1_FTMCHAN,
-#endif
-#if PWM_1_CHANNELS > 2
-    PWM_1_CH2_FTMCHAN,
-#endif
-#if PWM_1_CHANNELS > 3
-    PWM_1_CH3_FTMCHAN,
-#endif
-#if PWM_1_CHANNELS > 4
-    PWM_1_CH4_FTMCHAN,
-#endif
-#if PWM_1_CHANNELS > 5
-    PWM_1_CH5_FTMCHAN,
-#endif
-#if PWM_1_CHANNELS > 6
-    PWM_1_CH6_FTMCHAN,
-#endif
-#if PWM_1_CHANNELS > 7
-    PWM_1_CH7_FTMCHAN,
-#endif
-};
-#endif
+#define PRESCALER_MAX       (7U)
 
-uint32_t pwm_init(pwm_t dev, pwm_mode_t mode, uint32_t freq, uint16_t res)
+static inline FTM_Type *ftm(pwm_t pwm)
 {
-    FTM_Type *ftm;
-    int channels = 0;
-    uint32_t pwm_clk = 0;
-    const uint8_t *ftmchan = NULL;
-
-    switch (dev) {
-#if PWM_0_EN
-
-        case PWM_0:
-            channels = PWM_0_CHANNELS;
-            pwm_clk = PWM_0_CLK;
-            ftm = PWM_0_DEV;
-            ftmchan = &ftm0chan[0];
-            break;
-#endif
-#if PWM_1_EN
-
-        case PWM_1:
-            channels = PWM_1_CHANNELS;
-            pwm_clk = PWM_1_CLK;
-            ftm = PWM_1_DEV;
-            ftmchan = &ftm1chan[0];
-            break;
-#endif
-
-        default:
-            return 0;
-    }
-
-    switch (mode) {
-        case PWM_LEFT:
-        case PWM_RIGHT:
-        case PWM_CENTER:
-            break;
+    return pwm_config[pwm].ftm;
+}
 
-        default:
-            return 0;
-    }
+uint32_t pwm_init(pwm_t pwm, pwm_mode_t mode, uint32_t freq, uint16_t res)
+{
+    uint8_t pre = 0;
 
-    if ((unsigned int)res > (PWM_MAX_VALUE + 1) || (res * freq) > pwm_clk) {
+    if (pwm >= PWM_NUMOF || ((res * freq) > CLOCK_BUSCLOCK)) {
```

Comment:
IMHO that is unnecessary, as `pwm` is not an enum value (anymore). From pwm.h:
```c
#define PWM_DEV(x)          (x)
...
typedef unsigned int pwm_t;
```
So there should be no need for (ugly) casting here?!